### PR TITLE
feat(asdf): Add support for more ASDF plugins

### PR DESCRIPTION
**Pull request by RothAndrew**
## Changes

Add support for ASDF plugins:
- adr-tools
- checkov
- flux2
- golangci-lint
- hadolint
- kubectl
- pre-commit
- terraform-docs
- tflint
- tfsec

## Context

I'm a user of Renovate, ASDF, and this list of ASDF plugins that aren't (yet) supported by Renovate.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

image

image

### CODE REVIEW

File: lib/modules/manager/asdf/extract.spec.ts [hunk 1]
```diff
@@ -43,10 +43,12 @@ describe('modules/manager/asdf/extract', () => {
 
     it('can handle multiple tools in one file', () => {
       const res = extractPackageFile(
-        codeBlock`argocd 2.5.4
+        codeBlock`adr-tools 3.0.0
```
Comment (viceice) [inline lib/modules/manager/asdf/extract.spec.ts:46]:
```suggestion
        codeBlock`
adr-tools 3.0.0
```
`codeBlock` will trim the string

File: lib/modules/manager/asdf/extract.spec.ts [hunk 2]
```diff
@@ -184,6 +206,13 @@ dummy 1.2.3
             datasource: 'flutter-version',
             depName: 'flutter',
           },
+          {
+            currentValue: '0.41.2',
+            datasource: 'github-tags',
+            packageName: 'fluxcd/flux2',
+            depName: 'flux2',
+            extractVersion: '^v(?<version>\\S+)',
```
Comment (viceice) [inline lib/modules/manager/asdf/extract.spec.ts:214]:
```suggestion
            extractVersion: '^v(?<version>.+)',
```
i don't think git tags can have spaces?

Comment (RothAndrew) [reply lib/modules/manager/asdf/extract.spec.ts:214]: I have no problem changing it, though I did copy what most (all?) of the other blocks were using for `extractVersion`.

We could
- Change just this one
- Change all of them
- Leave it as `\\S`

I'm happy do whichever one is requested 😄

Comment (viceice) [reply lib/modules/manager/asdf/extract.spec.ts:215]: Change all new, we can change the existing in a seperate PR.

**Review by viceice — COMMENTED** [review]
see comment

**Review by viceice — APPROVED** [review]

**Pull request merged**